dev-util/electron/electron-1.6.15.ebuild

```diff
@@ -316,6 +321,48 @@ _get_target_arch() {
 	echo -n "${target_arch}"
 }
 
+_apply_gentoo_patches() {
```

Comment:
Why do you have to make it so complex? Now figuring out which patches actually get applied will take half an hour… Not to mention pointing out all the problems with the code that follows.

Reply:
This approach makes things much easier to maintain for me.  Most ebuilds that distribute patches in tarballs use `epatch`.  This is basically a variant of `epatch` that does two extra things: 

1. Uses `/etc/portage/patches`-like semantics to separate generic patches from version-specific patches.
2. Electron is a bunch of distinct codebases stitched together.  I maintain the patches as git branches, so they all are rooted differently, this function takes care of changing the directory before doing `eapply`.

> Not to mention pointing out all the problems with the code that follows.

Which problems?

Reply:
> 1. Uses `/etc/portage/patches`-like semantics to separate generic patches from version-specific patches.

And why does the ebuild need to care about that? Are you planning to distribute irrelevant patches in the tarball? Just distribute the correct set of patches, and apply the whole directory.

> 2. Electron is a bunch of distinct codebases stitched together.  I maintain the patches as git branches, so they all are rooted differently, this function takes care of changing the directory before doing `eapply`.

Then please use separate directories for those different codebases and explicitly hardcode calling `eapply` for each of those directories in ebuild. Then whoever has to touch the ebuild in the future wouldn't have to figure out all that horrible `__` magic.

Reply:
> And why does the ebuild need to care about that? Are you planning to distribute irrelevant patches in the tarball? 

I keep patches in a git repo and export the entire repo, so, yes, the tarball will include patches for all versions/slots that are supported.  Why is this a problem?

> Then please use separate directories for those different codebases and explicitly hardcode calling eapply for each of those directories in ebuild.

OK.

Reply:
Because it means the tarball size will keep growing (minus compression effectiveness) with useless data.

Reply:
> Because it means the tarball size will keep growing (minus compression effectiveness) with useless data.

Obviously, no-longer-needed patches will be dropped.

And then, seriously?  The total size of the Electron source tarballs is *540 megabytes*.  I'm pretty sure half of that stuff isn't even compiled.  The current size of the patch tarball is *28 kilobytes*.  _Maybe_ it'll get to 50 (it's unlikely we'll keep more than three slots simultaneously).  

Now, Michał, I _agree_ with your desire to ensure that the things are clean and maintainable.  But in this case, you're basically asking me to do extra work on top of an already enormous effort to package this thing.  I will have to spend time to carve out those extra few kilobytes, upload the tarball manually somewhere (instead of just relying on the github tarball export).